developers/blobstream-rollups.md

```diff
@@ -0,0 +1,399 @@
+```
+
+Finally, we can compare the computed index with the sequence of spans,
+and be sure that the data/shares is part of the rollup data.
+
+#### Pros
+
+- Using a sequence of spans instead of the share commitment allows for simpler proofs
+
+#### Cons
+
+None
+
+## Optimistic rollups
+
+One type of rollups that can be built with Blobstream is optimistic rollups.
+An optimistic rollup is a rollup that commits optimistically to a set of blocks,
+and allows the other parties to verify that the blocks are valid,
+and if they're not, they can create fraud proofs to signal that.
+
+Celestia allows optimistic rollups to post their data on its DA layer,
+and to prove that the data is available using Blobstream.
+
+To build an optimistic rollup that uses Celestia as a DA layer, the following constructions can be inspired by.
+
+### Optimistic rollups that uses a sequence of spans
+
+Optimistic rollups can post their data in Celestia, then in the rollup settlement contract,
+they can reference optimistically that data using a [sequence of spans](#sequence-of-spans).
+Then, rollup full nodes can verify if that data is valid.
+If not, they can trigger a fraud proof.
+
+When using a sequence of spans, triggering the data availability fraud proofs,
+which are different from the state transitions fraud proofs (left for the rollup to define),
+goes back to the following cases:
+
+- Proving that the rollup data is unavailable:
+  this goes back to [proving that the sequence of spans is out of bounds](#proving-unavailable-data).
+- Proving an invalid state transition: this goes back to:
+  - [Proving that the rollup data is available and is part of the sequence of spans](#proving-inclusion-of-some-data).
+  - Parsing that data and verifying the invalid state transition:
+    this is the rollup logic fraud proofs, and it's left to the rollup to define.
+
+#### Pros
+
+- Not needing to verify anything at the moment of submitting the commitments to the rollup settlement contracts
+- The fraud proofs are simple and can be reduced to a single share: If, for example,
+  a single transaction in the rollup data that was posted to Celestia is faulty,
+  only the shares containing that transaction,
+  which can be as minimal as a single share, need to be proven on chain and verified.
+
+#### Cons
+
+None
+
+#### Example
+
+An example optimistic rollup
+that uses sequence of spans
+to reference its data can be found in the [`RollupInclusionProofs`](https://github.com/celestiaorg/blobstream-contracts/blob/master/src/lib/verifier/test/RollupInclusionProofs.t.sol).
+It portrays the different possible data availability proofs, constructs them and shows how to verify them.
+
+Also,
+more details
+on querying these kinds of proofs can be found in the [proof queries](https://docs.celestia.org/developers/blobstream-proof-queries) documentation.
+
+### Optimistic rollups that use share commitments
```

Comment:
if there really are no cons the first scheme, will anyone use this scheme? if so, why? maybe list that as a con above. If not, then perhaps we can remove this section

Reply:
I want to include this construction because it's how current rollups think it should be done, then add cons for it